kavm/src/kavm/__main__.py

```diff
@@ -0,0 +1,225 @@
+    kavm_lib_dir = Path(str(os.environ.get('KAVM_LIB')))
+
+    if not teal_programs_parser:
+        teal_programs_parser = kavm_lib_dir / 'scripts/parse-teal-programs.sh'
+    if not avm_simulation_parser:
+        avm_simulation_parser = kavm_lib_dir / 'scripts/parse-avm-simulation.sh'
+    try:
+        proc_result = kavm.run_avm_simulation(
+            input_file=input_file,
+            output=output,
+            profile=profile,
+            teal_sources_dir=teal_sources_dir,
+            teal_programs_parser=teal_programs_parser,
+            avm_simulation_parser=avm_simulation_parser,
+        )
+        print(proc_result.stdout)
+        exit(proc_result.returncode)
+    except CalledProcessError as err:
+        exit(err.returncode)
+
+
+def main() -> None:
+    sys.setrecursionlimit(15000000)
+    parser = create_argument_parser()
+    args = parser.parse_args()
+    logging.basicConfig(level=_loglevel(args), format=_LOG_FORMAT)
+
+    if not args.definition_dir:
+        env_definition_dir = os.environ.get('KAVM_DEFINITION_DIR')
+        if env_definition_dir:
+            args.definition_dir = Path(env_definition_dir)
+        else:
+            raise RuntimeError('Cannot find KAVM definition, plese specify either --definition or KAVM_DEFINITION_DIR')
+
+    executor_name = 'exec_' + args.command.lower().replace('-', '_')
+    if executor_name not in globals():
+        raise AssertionError(f'Unimplemented command: {args.command}')
+
+    execute = globals()[executor_name]
+    execute(**vars(args))
+
+
+def create_argument_parser() -> ArgumentParser:
+    parser = ArgumentParser(prog='kavm')
+
+    shared_args = ArgumentParser(add_help=False)
+    shared_args.add_argument('--verbose', '-v', default=False, action='store_true', help='Verbose output.')
+    shared_args.add_argument('--debug', default=False, action='store_true', help='Debug output.')
+    shared_args.add_argument('--profile', default=False, action='store_true', help='Coarse process-level profiling.')
+
+    command_parser = parser.add_subparsers(dest='command', required=True, help='Command to execute')
+
+    # kompile
+    kompile_subparser = command_parser.add_parser('kompile', help='Kompile KAVM', parents=[shared_args])
+    kompile_subparser.add_argument(
+        '--definition-dir', dest='definition_dir', type=Path, help='Path to store the kompiled definition'
+    )
+    kompile_subparser.add_argument(
+        '-I', type=str, dest='includes', default=[], action='append', help='Directories to lookup K definitions in.'
+    )
+    kompile_subparser.add_argument('main_file', type=Path, help='Path to the main file')
+    kompile_subparser.add_argument('--main-module', type=str)
+    kompile_subparser.add_argument('--syntax-module', type=str)
+
+    # kast
+    kast_subparser = command_parser.add_parser('kast', help='Kast a term', parents=[shared_args])
+    kast_subparser.add_argument(
+        '--definition',
+        dest='definition_dir',
+        type=Path,
```

Comment:
Module `pyk.cli` contains functions `file_path` and `dir_path` that check if the corresponding path is an existing file or directory, respectively:

```suggestion
        type=dir_path,
```